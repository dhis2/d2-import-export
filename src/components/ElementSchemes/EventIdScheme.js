import React from 'react'
import PropTypes from 'prop-types'

import { SelectField } from '../'
import { eventIdSchemeOptions } from '../../utils/options'

const EventIdScheme = ({ name, label, dataTest }) => {
    return (
        <SelectField
            name={name}
            label={label}
            options={eventIdSchemeOptions}
            dense
            dataTest={dataTest}
        />
    )
}

EventIdScheme.propTypes = {
    dataTest: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
}

export { EventIdScheme }
