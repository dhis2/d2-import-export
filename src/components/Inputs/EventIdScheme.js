import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_UID = { value: 'UID', label: i18n.t('Uid') }
export const OPTION_CODE = { value: 'CODE', label: i18n.t('Code') }

export const EVENT_ID_SCHEME_KEY = 'eventIdScheme'
export const EVENT_ID_SCHEME_DEFAULT_VALUE = OPTION_UID.value

export const EventIdScheme = () => (
    <RadioGroup
        name={EVENT_ID_SCHEME_KEY}
        label={i18n.t('Event id scheme')}
        options={[OPTION_UID, OPTION_CODE]}
        dataTest="input-event-id-scheme"
    />
)
