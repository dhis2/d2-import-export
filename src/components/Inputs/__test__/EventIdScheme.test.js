import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { EventIdScheme } from '../EventIdScheme'

jest.mock('react-final-form')

describe('Input component - EventIdScheme', () => {
    it('should render correctly', () => {
        const file = shallow(<EventIdScheme />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
