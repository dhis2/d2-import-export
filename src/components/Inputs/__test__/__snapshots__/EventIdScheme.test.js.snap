// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - EventIdScheme should render correctly 1`] = `
<Field>
  <Select
    dataTest="input-event-id-scheme"
    label="Event id scheme"
    name="eventIdScheme"
    options={
      Array [
        Object {
          "label": "Uid",
          "value": "UID",
        },
        Object {
          "label": "Code",
          "value": "CODE",
        },
      ]
    }
  />
</Field>
`;
